equinix/resource_metal_device.go

```diff
@@ -655,10 +661,13 @@ func resourceMetalDeviceRead(ctx context.Context, d *schema.ResourceData, meta i
 		d.Set("storage", storageString)
 	}
 	if device.HardwareReservation != nil {
-		d.Set("deployed_hardware_reservation_id", device.HardwareReservation.ID)
+		d.Set("deployed_hardware_reservation_id", device.HardwareReservation.GetId())
```

Comment:
This may also be a case to simplify with `GetHardwareReservation().GetId()` (removing the `if` clause.

Reply:
The `if` clause here is not specific to the SDK conversion